docs/decisions/0008-secret-key-usage.rst

```diff
@@ -0,0 +1,37 @@
+Django Secret Key Usage
+-----------------------
+
+Status
+======
+
+Accepted
+
+Context
+=======
+
+Django's Secret key is used by django to cryptographically sign various
+capabilities in django.  It is also used by the edx-platform in various cases
+as a `pepper`_ when hashing data to anonymize it while keeping it consistent.
+Some of the uses of the ``SECRET_KEY`` in edx-platform were not resilient to
+the key being rotated.  As a part of ARCHBOM-1646 those code paths are being
+updated so that regular rotation of the ``SECRET_KEY`` will cause temporary
+issues but will not fully break things.  ie. Sessions might get invalidated or
+data might not be corelatable over the change boundry but the change is
+otherwise safe.
+
+.. _pepper: https://en.wikipedia.org/wiki/Pepper_(cryptography)
+
+Decisions
+=========
+
+1. Any uses of ``SECRET_KEY`` should be resilient to that key being rotated.
+
+2. Wherever the secret key is used, it should also document the impact of
+   rotating the key.
```

Comment:
```suggestion
2. Wherever the secret key is used, it should also document the impact of
   rotating the key or of exposure of the key to an attacker.
```

I'm not sure about the exact phrasing, but I'd like to request that people document the consequences of exposure as well.

Reply:
What do you think about just making that a separate  item.
```
3. Wherever the secret key is used, it should also document the consequences of the key being exposed.
```

Reply:
Yeah, I like that.

Reply:
Hmm, but also not just impact of rotating the key -- any process that's required, too.

Reply:
(I mean, really, we should have these for all secrets... separate ADR?)

Reply:
(I guess it would be an OEP, really.)